Python/Product/Uwp/Interpreter/PythonUwpInterpreterFactoryProvider.cs

```diff
@@ -23,39 +25,68 @@ namespace Microsoft.PythonTools.Uwp.Interpreter {
     class PythonUwpInterpreterFactoryProvider : IPythonInterpreterFactoryProvider {
         private HashSet<IPythonInterpreterFactory> _factories = null;
 
-        public const string DefaultVersion = "3.5";
-
         public event EventHandler InterpreterFactoriesChanged;
 
         public PythonUwpInterpreterFactoryProvider() {
-            PythonVersion = DefaultVersion;
-        }
-
-        public string PythonVersion {
-            get; set;
         }
 
         private void DiscoverFactories() {
             if (_factories == null) {
-                var defaultConfig = new InterpreterConfiguration(
-                    null,
-                    null,
-                    null,
-                    null,
-                    null,
-                    ProcessorArchitecture.None,
-                    new Version(PythonVersion),
-                    InterpreterUIMode.CannotBeDefault
-                );
+                var userSdkInstallDir = new DirectoryInfo(Environment.ExpandEnvironmentVariables(@"%LOCALAPPDATA%\Microsoft\VisualStudio\%VISUALSTUDIOVERSION%Exp\Extensions\Microsoft\Python UWP"));
+                var sdkInstallDir = new DirectoryInfo(Environment.ExpandEnvironmentVariables(@"%ProgramFiles(x86)%\Microsoft SDKs\Windows\v10.0\ExtensionSDKs\Python UWP"));
 
                 _factories = new HashSet<IPythonInterpreterFactory>();
 
-                _factories.Add(new PythonUwpInterpreterFactory(defaultConfig));
-
-                var factoriesChanged = InterpreterFactoriesChanged;
-
-                if (factoriesChanged != null) {
-                    factoriesChanged(this, new EventArgs());
+                if (userSdkInstallDir.Exists) {
+                    foreach (var dirInfo in userSdkInstallDir.EnumerateDirectories()) {
+                        var targetsFile = dirInfo.GetFiles("CPython.targets", SearchOption.AllDirectories).FirstOrDefault();
+
+                        if (targetsFile != null) {
+                            _factories.Add(
+                                new PythonUwpInterpreterFactory(
+                                    new InterpreterConfiguration(
+                                        dirInfo.FullName,
+                                        targetsFile.FullName,
+                                        null,
+                                        null,
+                                        null,
+                                        ProcessorArchitecture.None,
+                                        new Version(dirInfo.Name),

```

Comment:
This is brave, since anyone could make a `CPython.targets` file... `Version.TryParse` would be better, and if it's possible to rename `CPython.targets` to something more unique (even `CPython.UWP.targets` or `CPython.WinRT.targets` would be sufficient) then that'll help protect against rogue packages.